articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,340 @@
+
+#### Do-While Loop
+
+The do-While loop is an example of [Indefinite Loops](#indefinite-loops).
+
+**Syntax of Do-While Loop:**
+
+```Dart
+void main(){
+   // initialize counter variable
+
+   do {
+      //code;
+     
+      increment/decrement;
+   } while(condition);
+}
+```
+
+**How the Do-While Loop works:**
+
+**Code Execution - Increment - Condition Check**.
+
+1. Code executes then the counter variable initialized outside the syntax increments or decrements. 
+
+2. The condition check - if it's met, it begins from code execution. 
+
+3. The code execution will continue until the condition check turns false. 
+
+> Note that the code block in the Do-While loop executes at least once.
+
+***Example:***
+
+```Dart
+void main(){
+   int i = 0;
+   do {
+      print ("Hello");
+      i++;
+   } while (i<5);
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+
+The code first executes then the counter variable increments by one. It then checks the condition (1<5). It is true, so the loop will begin again. The code executes, then the counter variable increments by one. It then checks the condition (2<5). It meets the condition. It will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.
+
+### Other Key Concepts
+#### Break Keyword
+
+We use the `Break Keyword` to forcefully end a loop. It comes in handy when you want a partial output on a certain condtion and not to run all iterations.
+
+***Example:***
+
+```Dart
+void main(){
+ for (int i = 1; i<=10; i++){
+    print ("Hello");
+    if(i >= 6){
+       break;
+    }
+  }
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+This code will end the loop after printing five hellos because of the break statement.
+
+#### Continue Keyword
+
+We use the `Continue Keyword` to skip code or a loop iteration after the continuation and proceed with the rest of the iteration.
+
+***Example:***
+
+```Dart
+void main(){
+
+ for (int i = 1; i<=8; i++){
+   
+    print (i);
+
+    if(i == 5){
+      
+       continue;
+    }
+
+  }
+   
+}
+```
+Output:
+
+```
+1
+2
+3
+4
+5
+6
+7
+8
+```
+In the example, when i becomes 5, it skips the print statement, the rest of the loop continues.
+
+### Applications of Loops
+Here are some positive and negative applications of loops:
+
+1. **Writing a Music Player Code**: A music player has a loop that enables it to play from one song to the next systematically.
+
+2. **Cycling through values**: It is the use of loops to print out values in an extensive list. ***For example***, Names of students in a school.
+
+```Dart
+void main(){
+   
+   List grade-one-students = ["John", "Mark", "Alex"];
```

Comment:
You can't use `-` while naming variables. Please use underscore `_` or camelCase.